### PR TITLE
solves issue 3464 and half of issue 3466

### LINKED ISSUE #3464: `rpcuser` and `rpcpassword` MUST be set when using `BitcoindRpcAppConfig`
https://github.com/bitcoin-s/bitcoin-s/blob/1e87cb0fdecea0e67eb2679ee6c7b47ad644d72c/app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala#L88

This is not necessary. For instance if you want to authenticate against a locally running bitcoind instance via cookie authentication, these fields do _not_ need to be defined inside of the `bitcoin-s.conf` file. When using cookie based authentication (#2891) i should not need to specify `rpcuser` and `rpcpassword`. 

We need to modify the return types of `rpcuser` and `rpcpassword` to be `Option[String]` rather than `String` and then propagate the changes.

### PR DESCRIPTION
**Pull request by shruthii625**
fixes #3464 and part of #3466 

If rpcuser and rpcpassword are not set, then cookie based authentication will be used. 
If bitcoind is not found on PATH, bitcoin-s.conf will be read to obtain the binary.

### CODE REVIEW

File: app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala [hunk 1]
```diff
@@ -137,16 +147,25 @@ case class BitcoindRpcAppConfig(
...
       uri = uri,
       rpcUri = rpcUri,
       authCredentials = authCredentials,
       zmqConfig = zmqConfig,
-      binary = binaryOpt.getOrElse(fallbackBinary),
+      binary = binaryOpt match {
+        case Some(file) => file
+        case None => {
+          val homeVar = sys.env("HOME");
+          val config = ConfigFactory
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala]: Is there a reason this config reference doesn't work? 

https://github.com/bitcoin-s/bitcoin-s/blob/4f7b6422ea3390f276733ee3c0fb21d426bde751/app-commons/src/main/scala/org/bitcoins/commons/config/AppConfig.scala#L120

Comment (shruthii625) [reply app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala]: AppConfig already has a 'network' and a 'datadir'. Plus, if I extend AppConfig, I'd have to implement all of it's methods here. So, I wasn't too sure if I should use it

Comment (Christewart) [reply app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala]: It already extends `AppConfig`

https://github.com/shruthii625/bitcoin-s/blob/2fa4dddbcc673fd21c71ec55c09c00ab88f17da6/app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala#L25

Comment (shruthii625) [reply app/server/src/main/scala/org/bitcoins/server/BitcoindRpcAppConfig.scala]: I hadn't properly read your message and assumed you were talking about BitcoindInstance yikes
I've changed it to use the config of AppConfig and it seems to be working fine

**Pull request closed without merging**